projects/igniteui-angular/src/lib/dialog/dialog.component.ts

```diff
@@ -64,50 +64,181 @@ export class IgxDialogComponent implements IToggleView, OnInit, OnDestroy {
+     *<igx-dialog leftButtonColor="yellow" leftButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public leftButtonColor = '';
+
+    /**
+     * An @Input property to set the left button `background-color`.
+     *```html
+     *<igx-dialog leftButtonBackgroundColor="black" leftButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public leftButtonBackgroundColor = '';
+
+    /**
+     * An @Input property to set the left button `ripple`.
+     *```html
+     *<igx-dialog leftButtonRipple="green" leftButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public leftButtonRipple = '';
 
+    /**
+     * An @Input property to set the `label` of the right button of the dialog.
+     *```html
+     *<igx-dialog rightButtonLabel="OKAY" #alert title="Notification"  (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public rightButtonLabel = '';
+
+    /**
+     * An @Input property to set the right button `type`. The types are flat, raised and fab.
+     *```html
+     *<igx-dialog rightButtonType="fab" rightButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public rightButtonType = 'flat';
+
+    /**
+     * An @Input property to set the right button `color`.
+     *```html
+     *<igx-dialog rightButtonColor="yellow" rightButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public rightButtonColor = '';
+
+    /**
+     * An @Input property to set the right button `background-color`.
+     *```html
+     *<igx-dialog rightButtonBackgroundColor="black" rightButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public rightButtonBackgroundColor = '';
+
+    /**
+     * An @Input property to set the right button `ripple`.
+     *```html
+     *<igx-dialog rightButtonRipple="green" rightButtonLabel="OKAY" #alert (onLeftButtonSelect)="alert.close()"></igx-dialog>
+     *```
+     */
     @Input()
     public rightButtonRipple = '';
 
+    /**
+     * An @Input property that allows you to enable the "close on click outside the dialog". By default it's disabled.
+     *```html
+     *<igx-dialog closeOnOutsideSelect="true" #dialog leftButtonLabel="Cancel" (onLeftButtonSelect)="dialog.close()"
+     *rightButtonLabel="OK" rightButtonRipple="#4CAF50" (onRightButtonSelect)="onDialogOKSelected($event)">
+     *</igx-dialog>
+     *```
+     */
     @Input()
     public closeOnOutsideSelect = false;
 
+    /**
+     * An @Output property that triggers a function when the dialog is opened.
+     *```html
+     *<igx-dialog (onOpen)="alert.open()" #dialog leftButtonLabel="Cancel" (onLeftButtonSelect)="dialog.close()" rightButtonLabel="OK">
```

Comment:
Consider changing this code sample, you don't have `alert` in this context. It would be better to invoke a method from the component probably - (onOpen)="onDialogOpenHandler($event)"